src/rgw/rgw_common.cc

```diff
@@ -286,10 +286,17 @@ req_state::~req_state() {
 
 std::ostream& req_state::gen_prefix(std::ostream& out) const
 {
+  std::ios oldState(nullptr);
```

Comment:
there are already some attempts to reset the state in the code.
you can probably remove: `std::setprecision(p) << std::defaultfloat`